### ISSUE TITLE
Gn

### ISSUE THREAD

**Issue by Cvar1984** [termux/termux-packages#4777]
**Package description**
GN is a meta-build system that generates build files for Ninja.

**Link to home page and sources**
1. Home page:
2. Source code: https://gn.googlesource.com/gn

**Additional information**
I have try to compile on my device and i have some problem when i try to run it

ERROR at //.gn:5:1: Unable to load "/data/data/com.termux/files/home/v8-8.1.128/build/dotfile_settings.gni".
import("//build/dotfile_settings.gni")

**Comment by JCWasmx86** [comment]
I tried to build gn using the [instructions](https://gn.googlesource.com/gn).
It fails at `ninja -C out`, because of two things:
1. src/gn/function_write_file_unittest.cc (Around line 97): `ASSERT_EQ(futimes(foo_file.GetPlatformFile(), times), 0);`, futimes is not declared.
2. This command: `clang++ -O3 -fdata-sections -ffunction-sections -Wl,--gc-sections -Wl,-strip-all -Wl,--as-needed -static-libstdc++ -pthread -o gn -Wl,--start-group src/gn/gn_main.o base.a gn_lib.a -Wl,--end-group -ldl` and `clang++ -O3 -fdata-sections -ffunction-sections -Wl,--gc-sections -Wl,-strip-all -Wl,--as-needed -static-libstdc++ -pthread -o gn_unittests -Wl,--start-group src/gn/action_target_generator_unittest.o src/gn/analyzer_unittest.o src/gn/args_unittest.o src/gn/builder_unittest.o src/gn/c_include_iterator_unittest.o src/gn/command_format_unittest.o src/gn/commands_unittest.o src/gn/compile_commands_writer_unittest.o src/gn/config_unittest.o src/gn/config_values_extractors_unittest.o src/gn/escape_unittest.o src/gn/exec_process_unittest.o src/gn/filesystem_utils_unittest.o src/gn/file_writer_unittest.o src/gn/frameworks_utils_unittest.o src/gn/function_filter_unittest.o src/gn/function_foreach_unittest.o src/gn/function_forward_variables_from_unittest.o src/gn/function_get_label_info_unittest.o src/gn/function_get_path_info_unittest.o src/gn/function_get_target_outputs_unittest.o src/gn/function_process_file_template_unittest.o src/gn/function_rebase_path_unittest.o src/gn/function_template_unittest.o src/gn/function_toolchain_unittest.o src/gn/function_write_file_unittest.o src/gn/functions_target_rust_unittest.o src/gn/functions_target_unittest.o src/gn/functions_unittest.o src/gn/hash_table_base_unittest.o src/gn/header_checker_unittest.o src/gn/inherited_libraries_unittest.o src/gn/input_conversion_unittest.o src/gn/json_project_writer_unittest.o src/gn/rust_project_writer_unittest.o src/gn/rust_project_writer_helpers_unittest.o src/gn/label_pattern_unittest.o src/gn/label_unittest.o src/gn/loader_unittest.o src/gn/metadata_unittest.o src/gn/metadata_walk_unittest.o src/gn/ninja_action_target_writer_unittest.o src/gn/ninja_binary_target_writer_unittest.o src/gn/ninja_build_writer_unittest.o src/gn/ninja_bundle_data_target_writer_unittest.o src/gn/ninja_c_binary_target_writer_unittest.o src/gn/ninja_copy_target_writer_unittest.o src/gn/ninja_create_bundle_target_writer_unittest.o src/gn/ninja_generated_file_target_writer_unittest.o src/gn/ninja_group_target_writer_unittest.o src/gn/ninja_rust_binary_target_writer_unittest.o src/gn/ninja_target_command_util_unittest.o src/gn/ninja_target_writer_unittest.o src/gn/ninja_toolchain_writer_unittest.o src/gn/operators_unittest.o src/gn/output_conversion_unittest.o src/gn/parse_tree_unittest.o src/gn/parser_unittest.o src/gn/path_output_unittest.o src/gn/pattern_unittest.o src/gn/runtime_deps_unittest.o src/gn/scope_per_file_provider_unittest.o src/gn/scope_unittest.o src/gn/setup_unittest.o src/gn/source_dir_unittest.o src/gn/source_file_unittest.o src/gn/string_atom_unittest.o src/gn/string_output_buffer_unittest.o src/gn/string_utils_unittest.o src/gn/substitution_pattern_unittest.o src/gn/substitution_writer_unittest.o src/gn/target_unittest.o src/gn/template_unittest.o src/gn/test_with_scheduler.o src/gn/test_with_scope.o src/gn/tokenizer_unittest.o src/gn/unique_vector_unittest.o src/gn/value_unittest.o src/gn/vector_utils_unittest.o src/gn/version_unittest.o src/gn/visibility_unittest.o src/gn/visual_studio_utils_unittest.o src/gn/visual_studio_writer_unittest.o src/gn/xcode_object_unittest.o src/gn/xml_element_writer_unittest.o src/util/test/gn_test.o base.a gn_lib.a -Wl,--end-group -ldl` is failing with:

```
/data/data/com.termux/files/usr/bin/aarch64-linux-android-ld: cannot find -lc++_shared
```

Then I simply (temporarily) removed `-static-libstdc++` from the options in out/build.ninja.

Now it compiles and `out/gn_unittests` passes all 624 tests and the examples work, too.

**Comment by xtkoba** [comment]
Recently I solved a problem of `futimes` not being available in another package <a href="https://github.com/termux/termux-packages/blob/d617bb28bc4030bcfb48780a0befaf442b68f3fd/packages/uwsgi/sys_time.c">by backporting from newer bionic codebase</a>. I believe this works for `gn` too.

**Comment by sk0kanik** [comment]
It is available from #7290
